hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/net/NioInetPeer.java

```diff
@@ -91,7 +91,11 @@ public void close() throws IOException {
     try {
       in.close();
     } finally {
-      out.close();
+      try {
+        out.close();
+      } finally {
+        socket.close();
+      }
```

Comment:
@LiuGuH Hi, Leave one comment. Please check DFSUtilClient#peerFromSocket. 
https://github.com/apache/hadoop/blob/897f446d54315b70754eb8023cf195c902bca98e/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSUtilClient.java#L799
Should we better set keep-alive time here or not ?

Reply:
This is used for server side.  Keepalive is not a good idea.